src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -480,6 +513,77 @@ private ClusterDescription ProcessShardedChange(ClusterDescription clusterDescri
             return clusterDescription.WithServerDescription(args.NewServerDescription);
         }
 
+        private ClusterDescription ProcessStandaloneChange(ClusterDescription clusterDescription, ServerDescriptionChangedEventArgs args)
+        {
+            if (args.NewServerDescription.Type != ServerType.Standalone)
+            {
+                return RemoveServer(clusterDescription, args.NewServerDescription.EndPoint, "Server is not a standalone server.");
+            }
+
+            foreach (var endPoint in clusterDescription.Servers.Select(s => s.EndPoint).ToList())
+            {
+                if (!EndPointHelper.Equals(endPoint, args.NewServerDescription.EndPoint))
+                {
+                    clusterDescription = RemoveServer(clusterDescription, endPoint, "Removing all other end points once a standalone is discovered.");
+                }
+            }
+
+            return clusterDescription.WithServerDescription(args.NewServerDescription);
+        }
+
+        void IDnsMonitoringCluster.ProcessDnsException(Exception exception)
+        {
+            lock (_updateClusterDescriptionLock)
+            {
+                var newClusterDescription = Description.WithDnsMonitorException(exception);
```

Comment:
is this, and if so should this, rely on DnsMonitor not to call this method after calling processDnsResults?

Reply:
I'm not sure how important it is to ignore any exceptions that occur after the first successful DNS lookup, but in any case the logic to only call ProcessDnsException for exceptions that occur before the first successful DNS lookup in in the DnsMonitor.